pkg/controller/hyperconverged/hyperconverged_controller_test.go

```diff
@@ -2,230 +2,993 @@ package hyperconverged
-				Expect(cr.Spec.LinuxBridge).To(Equal(&networkaddons.LinuxBridge{}))
-				Expect(cr.Spec.KubeMacPool).To(Equal(&networkaddons.KubeMacPool{}))
+			It("should find if present", func() {
+				hco := &hcov1alpha1.HyperConverged{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: namespace,
+					},
+					Spec: hcov1alpha1.HyperConvergedSpec{},
+				}
+
+				expectedResource := newKubeVirtForCR(hco, namespace)
+				expectedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", expectedResource.Namespace, expectedResource.Name)
+				cl := initClient([]runtime.Object{hco, expectedResource})
+				r := initReconciler(cl)
+				Expect(r.ensureKubeVirt(hco, log, request)).To(BeNil())
+
+				// Check HCO's status
+				Expect(hco.Status.RelatedObjects).To(Not(BeNil()))
+				objectRef, err := reference.GetReference(r.scheme, expectedResource)
+				Expect(err).To(BeNil())
+				// ObjectReference should have been added
+				Expect(hco.Status.RelatedObjects).To(ContainElement(*objectRef))
+				// Check conditions
+				Expect(r.conditions).To(ContainElement(RepresentCondition(conditionsv1.Condition{
+					Type:    conditionsv1.ConditionAvailable,
+					Status:  corev1.ConditionFalse,
+					Reason:  "KubeVirtConditions",
+					Message: "KubeVirt resource has no conditions",
+				})))
+				Expect(r.conditions).To(ContainElement(RepresentCondition(conditionsv1.Condition{
+					Type:    conditionsv1.ConditionProgressing,
+					Status:  corev1.ConditionTrue,
+					Reason:  "KubeVirtConditions",
+					Message: "KubeVirt resource has no conditions",
+				})))
+				Expect(r.conditions).To(ContainElement(RepresentCondition(conditionsv1.Condition{
+					Type:    conditionsv1.ConditionUpgradeable,
+					Status:  corev1.ConditionFalse,
+					Reason:  "KubeVirtConditions",
+					Message: "KubeVirt resource has no conditions",
+				})))
+			})
+
+			It("should handle conditions", func() {
+				hco := &hcov1alpha1.HyperConverged{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: namespace,
+					},
+					Spec: hcov1alpha1.HyperConvergedSpec{},
+				}
+
+				expectedResource := newKubeVirtForCR(hco, namespace)
+				expectedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", expectedResource.Namespace, expectedResource.Name)
+				expectedResource.Status.Conditions = []kubevirtv1.KubeVirtCondition{
+					kubevirtv1.KubeVirtCondition{
+						Type:    kubevirtv1.KubeVirtConditionAvailable,
+						Status:  corev1.ConditionFalse,
+						Reason:  "Foo",
+						Message: "Bar",
+					},
+					kubevirtv1.KubeVirtCondition{
+						Type:    kubevirtv1.KubeVirtConditionProgressing,
+						Status:  corev1.ConditionTrue,
+						Reason:  "Foo",
+						Message: "Bar",
+					},
+					kubevirtv1.KubeVirtCondition{
+						Type:    kubevirtv1.KubeVirtConditionDegraded,
+						Status:  corev1.ConditionTrue,
+						Reason:  "Foo",
+						Message: "Bar",
+					},
+				}
```

Comment:
Here it is testing one combination of conditions. Any thoughts about testing the other possible combinations in https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/docs/conditions.md?

Reply:
My focus in this pass was to cover all of the different paths through the `Reconcile()` it may be worth going through all of the possible combinations.